src/steamship/agents/utils.py

```diff
@@ -0,0 +1,19 @@
+from typing import Optional
+
+from steamship.agents.schema import AgentContext
+from steamship.agents.schema.llm import LLM
+
+_LLM_KEY = "llm"
+
+
+def with_llm(llm: LLM, context: Optional[AgentContext] = None) -> AgentContext:
```

Comment:
From the PR message it sounds like this felt mostly like an odd shim to write, but I actually think this style may prove super useful as a way to think....

If things like "Which LLM am I using" become dynamic settings of an instance-in-progress (e.g. 'let's try Cohere out'), then this kind of chaining could let us pile on user settings atop reasonable defaults in a pretty elegant way